src/borg/archiver.py

```diff
@@ -433,6 +435,14 @@ def item_is_hardlink_master(item):
             return (partial_extract and stat.S_ISREG(item.mode) and
                     item.get('hardlink_master', True) and 'source' not in item)
 
+        if progress:
+            extracted_size = 0
+            for item in archive.iter_items(filter=lambda item: matcher.match(item.path)):
+                extracted_size += item.file_size()
+            pi = ProgressIndicatorPercent(total=extracted_size, msg='Extracting files %5.1f%%', step=0.1, same_line=True)
+        else:
+            pi = None
```

Comment:
borg currently only does single passes for creation and extraction. this would be the first place that introduces a 2nd pass just for cosmetics.
I'ld suggest the progress indication be like in "create" and not use a 2nd pass.